iphone/Maps/Core/TextToSpeech/MWMTextToSpeech.mm

```diff
@@ -279,6 +282,43 @@ - (AVAudioPlayer *)audioPlayer {
   return _audioPlayer;
 }
 
+- (void)playRandomTestString {
+  if (![self isValid])
+    [self createVoice:[[self class] savedLanguage]];
+  
+  if (self.testStrings == nil)
+    return;
+  
+  [self speakOneString: self.testStrings[_testStringIndex]];
+  
+  _testStringIndex++;
+  if (_testStringIndex >= self.testStrings.count)
+    _testStringIndex = 0;
+}
+
+- (NSArray *)testStrings {
+  NSString * language = [NSString stringWithUTF8String: bcp47ToTwineLanguage([[self class] savedLanguage]).c_str()];
+  NSString * languagePath = [NSBundle.mainBundle pathForResource: language ofType: @"lproj"];
+  if (languagePath == nil) {
+    LOG(LWARNING,
+        ([@"Couldn't find translation file for " stringByAppendingString: language]));
+    return nil;
+  }
+  NSBundle * bundle = [NSBundle bundleWithPath: languagePath];
+  
+  NSMutableArray * appTips = [NSMutableArray new];
+  NSString * notFound = @"_not found";
+  for(int idx = 0; ; idx++) {
+    NSString * appTipKey = [NSString stringWithFormat:@"app_tip_%02d", idx];
+    NSString * appTip = [bundle localizedStringForKey:appTipKey value:notFound table:nil];
+    if ([appTip isEqualToString: notFound])
+      break;
+    [appTips addObject: appTip];
```

Comment:
nit: some random shuffling after making an array would be cool.

Reply:
NSArray doesn't have a built in shuffle method so I used one from stackoverflow.